src/common/ceph_json.h

```diff
@@ -126,29 +298,47 @@ class JSONDecoder {
   JSONParser parser;
 
   JSONDecoder(ceph::buffer::list& bl) {
-    if (!parser.parse(bl.c_str(), bl.length())) {
-      std::cout << "JSONDecoder::err()" << std::endl;
+    if (!parser.parse(bl.c_str(), bl.length()))
       throw JSONDecoder::err("failed to parse JSON input");
-    }
   }
 
   template<class T>
-  static bool decode_json(const char *name, T& val, JSONObj *obj, bool mandatory = false);
+  static bool decode_json(std::string_view name, T& val, JSONObj *obj, bool mandatory = false);
 
   template<class C>
-  static bool decode_json(const char *name, C& container, void (*cb)(C&, JSONObj *obj), JSONObj *obj, bool mandatory = false);
+  static bool decode_json(std::string_view name, C& container, void (*cb)(C&, JSONObj *obj), JSONObj *obj, bool mandatory = false);
 
   template<class T>
-  static void decode_json(const char *name, T& val, const T& default_val, JSONObj *obj);
+  static void decode_json(std::string_view name, T& val, const T& default_val, JSONObj *obj);
 
   template<class T>
-  static bool decode_json(const char *name, boost::optional<T>& val, JSONObj *obj, bool mandatory = false);
+  static bool decode_json(std::string_view name, boost::optional<T>& val, JSONObj *obj, bool mandatory = false);
 
   template<class T>
-  static bool decode_json(const char *name, std::optional<T>& val, JSONObj *obj, bool mandatory = false);
-
+  static bool decode_json(std::string_view name, std::optional<T>& val, JSONObj *obj, bool mandatory = false);
 };
 
+template <typename IntegerT>
+requires ceph_json::detail::json_signed_integer<IntegerT> ||
+         ceph_json::detail::json_unsigned_integer<IntegerT>
+void decode_json_obj(IntegerT& val, JSONObj *obj)
+try
+{
+ if constexpr (ceph_json::detail::json_signed_integer<IntegerT>)
+  val = stol(obj->get_data());
+
+ if constexpr (ceph_json::detail::json_unsigned_integer<IntegerT>)
+  val = stoul(obj->get_data());
```

Comment:
won't this overload match 64-bit ints which would need `stoll()`/`stoull()`? consider using @adamemerson's [ceph::parse()](https://github.com/ceph/ceph/blob/9fa163d/src/common/strtol.h#L39-L41) from common/strtol.h:
```c++
std::optional result = ceph::parse<IntegerT>(obj->get_data());
if (result) {
  val = *result;
} else { // throw
```

Reply:
I believe it would, indeed-- and although it should throw a range error, that doesn't mean I'm checking for one! I'll look at ceph::parse(), thank you!